graphrag/index/__main__.py

```diff
@@ -63,32 +63,23 @@
         help="Create an initial configuration in the given path.",
         action="store_true",
     )
-    parser.add_argument(
-        "--overlay-defaults",
```

Comment:
Check with Josh, since they use this param. This means overlay will always happen right? Which is totally fine, not overlaying was an old behaviour

Reply:
I found the overlay logic confusing. Before merging the indexing API, from what I could tell, a default config was always being loaded and merged somewhere in the stack. When the indexing API was merged we had conversations about simplifying the logic around it. When I removed the option and unit and smoke tests still passed I figured my assumption was correct but it may not have been captured well in the tests. Thoughts on removing this option @jgbradley1?